operator/pkg/networkpolicy/cell.go

```diff
@@ -0,0 +1,236 @@
+	))
+}
+
+func (pv *policyValidator) handleCNPEvent(ctx context.Context, event resource.Event[*cilium_api_v2.CiliumNetworkPolicy]) error {
+	var err error
+	defer func() {
+		event.Done(err)
+	}()
+	if event.Kind != resource.Upsert {
+		return nil
+	}
+
+	pol := event.Object
+	log := pv.params.Logger.WithFields(logrus.Fields{
+		logfields.K8sNamespace:            pol.Namespace,
+		logfields.CiliumNetworkPolicyName: pol.Name,
+	})
+
+	var errs error
+	if pol.Spec != nil {
+		errs = errors.Join(pol.Spec.Sanitize())
+	}
+	for _, r := range pol.Specs {
+		errs = errors.Join(r.Sanitize())
+	}
+
+	newPol := pol.DeepCopy()
+	newPol.Status.Conditions = updateCondition(event.Object.Status.Conditions, errs)
+	if newPol.Status.DeepEqual(&pol.Status) {
+		return nil
+	}
+
+	if errs != nil {
+		log.WithField(logfields.Error, errs).Debug("Detected invalid CNP, setting condition")
+	} else {
+		log.Debug("CNP now valid, setting condition")
+	}
+	// Using the UpdateStatus subresource will prevent the generation from being bumped.
+	_, err = pv.params.Clientset.CiliumV2().CiliumNetworkPolicies(pol.Namespace).UpdateStatus(
+		ctx,
+		newPol,
+		metav1.UpdateOptions{},
+	)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		log.WithError(err).Error("failed to update CNP status")
+	}
+
+	return err
+}
+
+func (pv *policyValidator) handleCCNPEvent(ctx context.Context, event resource.Event[*cilium_api_v2.CiliumClusterwideNetworkPolicy]) error {
+	var err error
+	defer func() {
+		event.Done(err)
+	}()
+	if event.Kind != resource.Upsert {
+		return nil
+	}
+
+	pol := event.Object
+	log := pv.params.Logger.WithFields(logrus.Fields{
+		logfields.K8sNamespace:                       pol.Namespace,
+		logfields.CiliumClusterwideNetworkPolicyName: pol.Name,
+	})
+
+	var errs error
+	if pol.Spec != nil {
+		errs = errors.Join(pol.Spec.Sanitize())
+	}
+	for _, r := range pol.Specs {
+		errs = errors.Join(r.Sanitize())
+	}
+
+	newPol := pol.DeepCopy()
+	newPol.Status.Conditions = updateCondition(event.Object.Status.Conditions, errs)
+	if newPol.Status.DeepEqual(&pol.Status) {
+		return nil
+	}
+
+	if errs != nil {
+		log.WithField(logfields.Error, errs).Debug("Detected invalid CCNP, setting condition")
+	} else {
+		log.Debug("CCNP now valid, setting condition")
+	}
+	// Using the UpdateStatus subresource will prevent the generation from being bumped.
+	_, err = pv.params.Clientset.CiliumV2().CiliumClusterwideNetworkPolicies().UpdateStatus(
+		ctx,
+		newPol,
+		metav1.UpdateOptions{},
+	)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		log.WithError(err).Error("failed to update CCNP status")
+	}
+
+	return err
+}
+
+// updateCondition creates or updates the policy validation condition in Conditions, setting
+// the transition time as necessary.
+func updateCondition(conditions []cilium_api_v2.NetworkPolicyCondition, errs error) []cilium_api_v2.NetworkPolicyCondition {
+	wantCondition := corev1.ConditionFalse
```

Comment:
Shouldn't this be `ConditionTrue` to clear the condition once errs is nil?

Reply:
argh! good catch.